php/class-plugin.php

```diff
@@ -37,6 +37,7 @@ public function __construct() {
 	function init() {
 		add_action( 'wp_default_scripts', array( $this, 'register_scripts' ), 11 );
 		add_action( 'wp_default_styles', array( $this, 'register_styles' ), 11 );
+		add_action( 'admin_init', array( $this, 'add_caps' ) );
```

Comment:
I _think_ that instead of using `admin_init` to add this role to the persistent role object that it could be better to just filter `user_has_cap` dynamically, for instance:

```php
<?php
/* ... */

    add_filter( 'user_has_cap', array( $this, 'filter_user_has_cap' ), 9, 4 );
}

/* ... */

/**
 * Add the customize_publish capability to users who can edit_theme_options by default.
 *
 * @param array   $allcaps An array of all the user's capabilities.
 * @param array   $caps    Actual capabilities for meta capability.
 * @param array   $args    Optional parameters passed to has_cap(), typically object ID.
 * @param WP_User $user    The user object.
 */
function filter_user_has_cap( $allcaps, $caps, $args, $user ) {
	unset( $caps, $args, $user );
	if ( ! empty( $allcaps['customize'] ) && ! empty( $allcaps['edit_theme_options'] ) ) {
		$allcaps['edit_theme_options'] = 'customize_publish';
	}
	return $allcaps;
}
```

What do you think? This would mean the DB wouldn't have to be manipulated, and I recall from Nacin's talk a couple  years ago that filters are the best bet for adding caps to a role:

http://wordpress.tv/2013/08/10/andrew-nacin-current-user-can-watch-this-talk/
http://mannieschumpert.com/blog/wordpress-capabilities-magic-with-map_meta_cap/
https://gist.github.com/mannieschumpert/8886289